pyt/__main__.py

```diff
@@ -39,67 +57,73 @@ def main(command_line_args=sys.argv[1:]):
     elif args.trim_reassigned_in:
         ui_mode = UImode.TRIM
 
-    path = os.path.normpath(args.filepath)
-
-    if args.ignore_nosec:
-        nosec_lines = set()
-    else:
-        file = open(path, 'r')
-        lines = file.readlines()
-        nosec_lines = set(
-            lineno for
-            (lineno, line) in enumerate(lines, start=1)
-            if '#nosec' in line or '# nosec' in line
-        )
+    files = discover_files(
+        args.targets,
+        args.excluded_paths,
+        args.recursive
+    )
 
-    if args.project_root:
-        directory = os.path.normpath(args.project_root)
-    else:
-        directory = os.path.dirname(path)
-    project_modules = get_modules(directory)
-    local_modules = get_directory_modules(directory)
+    for path in files:
```

Comment:
Before this `for` loop, you can have a `vulnerabilities = list()`, and then do

```python
vulnerabilities.append(find_vulnerabilities(
    cfg_list,
    ui_mode,
    args.blackbox_mapping_file,
    args.trigger_word_file,
    nosec_lines
))
```

Reply:
I think it seems bad
![selection_102](https://user-images.githubusercontent.com/19760132/41208523-31ba02a6-6d2d-11e8-949b-e2a69e0887e0.png)

Reply:
Does it find vulnerabilities in all the files, or just the last file (i.e. last iteration of the loop)? If I'm reading it write I think it might do e.g. `vulnerabilites = kitmap_vulns`...then finally `vulnerabilities = a.py_vulns`, and only report the last list.

Reply:
(As an aside, it seems strange it's not printing out the vulnerability info, and just seems to print the object.)

Reply:
You are right,  it taking last item on the list. have you idea for fix?

Reply:
I think the fix is having a list outside of the for loop and adding the vulnerabilities of each file to it. The code from my first comment should do it, although now that I think about it it's probably extend and not append.

Reply:
^Ahh, this was it! 👍 Just change `append` to `extend` and it'll all work! :D